zio-http/shared/src/main/scala/zio/http/endpoint/AuthType.scala

```diff
@@ -15,6 +15,13 @@ sealed trait AuthType { self =>
       .asInstanceOf[
         AuthType { type ClientRequirement = ClientReq },
       ]
+
+  private var authScopes: Option[List[String]]  = scala.None
```

Comment:
There is no need to have a var here. Please create a new subtype of AuthType that can hold a list of scopes and a auth type.